internal/controllers/gateway/route_utils.go

```diff
@@ -638,3 +645,64 @@ func sameCondition(a, b metav1.Condition) bool {
 		a.Reason == b.Reason &&
 		a.Message == b.Message
 }
+
+func setRouteParentStatusCondition[T types.ParentStatusT](parentStatus T, newCondition metav1.Condition) bool {
+	var conditionFound, changed bool
+	switch p := (any)(parentStatus).(type) {
+	case *gatewayv1beta1.RouteParentStatus:
+		for i, condition := range p.Conditions {
+			if condition.Type == newCondition.Type {
+				conditionFound = true
+				if condition.Status != newCondition.Status ||
+					condition.Reason != newCondition.Reason ||
+					condition.Message != newCondition.Message {
```

Comment:
What do you think about factorizing this condition as a function to make it available to the below check (gatewayv1alpha2) as well?

Reply:
This part is not read only, so I am afraid that there is not a fast and clear way to extract it.

Reply:
I meant something like

```go
func conditionNeedsUpdate(oldCondition, newCondition metav1.Condition) bool {
  return oldCondition.Status != newCondition.Status ||
             oldCondition.Reason != newCondition.Reason ||
	     oldCondition.Message != newCondition.Message
}
```

Reply:
updated. used `sameCondition` function here to reduce duplicate code.